lingua_franca/internal.py

```diff
@@ -456,10 +458,19 @@ def _call_localized_function(func, *args, **kwargs):
             lang_param_index = func_params.index('lang')
             full_lang_code = None
 
+            # Check if we need to add timezone awareness to any datetime object
+            if config.inject_timezones:
+                for k, v in kwargs.items():
+                    if isinstance(v, datetime):
+                        kwargs[k] = to_local(v)
```

Comment:
This is probably covered better in the overall review comment.

Reply:
the idea is to convert only tz naive datetime objects, not convert every date.

good catch on this mistake, all unittests use args instead of kwargs so i missed this